Figaro/src/main/scala/com/cra/figaro/experimental/structured/ComponentCollection.scala

```diff
@@ -30,6 +30,12 @@ import com.cra.figaro.algorithm.factored.factors.Variable
  * To create a new component for an element, you need to say what problem it belongs to.
  */
 class ComponentCollection {
+  /** Indicates whether the old method of creating chain factors should be used.
+    * This defaults to false since the new method is generally better, but it's
+    * useful to have it available for comparison.
+    */
+  var useOldChainMethod = false
```

Comment:
Should we consider making a 'ComponentCollection' trait or base class, and extending it with subclasses for the new and old behavior? Then, we could replace setting this variable with instantiation of the appropriate subclass.